nbconvert/exporters/slides.py

```diff
@@ -1,28 +1,78 @@
 """HTML slide show Exporter class"""
 
-#-----------------------------------------------------------------------------
-# Copyright (c) 2013, the IPython Development Team.
-#
+# Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
-#
-# The full license is in the file COPYING.txt, distributed with this software.
-#-----------------------------------------------------------------------------
 
-#-----------------------------------------------------------------------------
-# Imports
-#-----------------------------------------------------------------------------
+from copy import deepcopy
 
-from nbconvert import preprocessors
 from traitlets.config import Config
+from traitlets import Unicode
 
 from .html import HTMLExporter
 
-#-----------------------------------------------------------------------------
-# Classes
-#-----------------------------------------------------------------------------
+def prepare(nb):
+    """Add some convenenience metadata on cells for the slide template.
+    """
+    nb = deepcopy(nb)
+
+    for cell in nb.cells:
+        # Make sure every cell has a slide_type
+        cell.metadata.slide_type = cell.metadata.get('slideshow', {}).get('slide_type', '-')
+
+    # Find the first visible cell
+    for index, cell in enumerate(nb.cells):
+        if cell.metadata.slide_type not in {'notes', 'skip'}:
+            cell.metadata.slide_type = 'slide'
+            cell.metadata.slide_start = True
+            cell.metadata.subslide_start = True
+            first_slide_ix = index
+            break
+    else:
+        raise ValueError("All cells are hidden, cannot create slideshow")
+
+    in_fragment = False
+
+    for index, cell in enumerate(nb.cells[first_slide_ix+1:],
```

Comment:
jeje... at overall, I made the same in my branch :wink:, detecting the first slide and enumerating from that point (well the point+1 in this case)